core/os/os.cpp

```diff
@@ -494,7 +494,13 @@ int OS::get_power_percent_left() {
 	return -1;
 }
 
+void *OS::get_stack_bottom() const {
+	return _stack_bottom;
+}
+
 OS::OS() {
+	void *volatile stack_bottom;
```

Comment:
I'm sure that platform-specific main() procedures do not use the stack bottom and frames that could use it will be below that pointer on the stack.

The address at the beginning of each main() would be an "honest" stack bottom, but it would pollute the procedure on each platform plus you would need to remember to set it when adding a new platform. I like the uniform solution more here, because it is clean and should work well for any real use case.